### PR TITLE
feat: Simplify build-system by transitioning to pure Python wheel

### PR DESCRIPTION
**Pull request by jcfr**
Introduce dependency on the `s5cmd` package to use platform-specific wheels and streamline the build process by switching the build-backend from "scikit-build-core" to "hatchling".

File updates were done based on the output of from the `scientific-python/cookie` template after selecting the recommended backend, Hatchling, for pure Python projects.

Additionally, disable submission of coverage reports as the project is not explicitly registered.

### CODE REVIEW

**Comment by jcfr** [comment]
Since I would also like to address the packaging of CSV file at wheel build time described in https://github.com/ImagingDataCommons/idc-index/issues/41, lets's wait for a new release of scikit-build-core integrating https://github.com/scikit-build/scikit-build-core/pull/655

This will allow use to have very simple `CMakeLists.txt` only responsible to download the relevant CSV file and ensure it is distributed along side code.

Alternatively, we could consider creating a python package called `idc-index-data` with a date based versioning scheme. Similar to `certifi`[^1]

This would avoid bloating the python package index with duplicate of the index data each time there is a release of this package (especially important if we anticipate that this package will have more release than the data we would like to distribute)


[^1]: https://pypi.org/project/certifi/

References:
* https://github.com/ImagingDataCommons/idc-index/issues/41
* https://github.com/scikit-build/scikit-build-core/pull/655
* https://github.com/ImagingDataCommons/idc-index/issues/49

**Comment by fedorov** [comment]
@jcfr thank you for your contribution!

Do you plan to maintain the `s5cmd` python package under your personal account going forward?


> Alternatively, we could consider creating a python package called idc-index-data with a date based versioning scheme.
> ...
> This would avoid bloating the python package index with duplicate of the index data each time there is a release of this package (especially important if we anticipate that this package will have more release than the data we would like to distribute)

I think this would be the preferred approach. We definitely expect that there will be more software releases than binary attachment updates. We also expect that there will be more such binary attachments in the future, and also that the size of the individual files will grow together with the amount of data available in IDC.

**Comment by jcfr** [comment]
> Do you plan to maintain the s5cmd python package under your personal account going forward?

Since we didn't hear back from the owner of https://github.com/peak organization didn't get back to us, I would like to :arrow_right: transfer the project to https://github.com/ImagingDataCommons. This will streamline the maintenance.


> > could consider creating a python package called idc-index-data

> preferred approach

This is now described in https://github.com/ImagingDataCommons/idc-index/issues/49

Once you created the project `idc-index-data` and grant me access, I can take care of setting it up along with updating this project.

File: idc_index/idc_index_data/__init__.py
Comment (jcfr) [inline idc_index/idc_index_data/__init__.py:10]: This sub-package illustrates what the standalone `idc-index-data` will look like.

File: idc_index/index.py
Comment (jcfr) [inline idc_index/index.py:20]: After we have a standalone `idc-index-data` package versioned on its own and available through PyPI, this will be changed to `import idc_index_data`

File: idc_index/idc_index_data/__init__.py
Comment (jcfr) [reply idc_index/idc_index_data/__init__.py:10]:
```suggestion
__version__ = "17.0"
```

I suggest we start versioning `idc-index-data` with `17.0` with `17` being the version of the `bigquery-public-data.idc_current.dicom_all_view` table, and `0` being the minor version we can use for maintenance fixes specific to the `idc-index-data` project.

**Comment by fedorov** [comment]
> I would like to ➡️ transfer the project to https://github.com/ImagingDataCommons. This will streamline the maintenance.

@jcfr I created a blank repo and invited you as Admin: https://github.com/ImagingDataCommons/s5cmd-python-distributions

I will follow up on other comments a bit later. We greatly appreciate your contributions!

**Comment by jcfr** [comment]
Thanks, it turns out that transferring will be much easier. 


I then suggest the following: (1) delete the project you just created, (2) add me to the IDC organization, (3) then once I accept the invitation, I should be able to transfer the project. 

You will then be able to accept it the request.

More details about the process are available in this issue where we went through a similar process for another project.
See https://github.com/scikit-build/scikit-build/pull/574#issuecomment-896461967

File: idc_index/idc_index_data/__init__.py
Comment (fedorov) [reply idc_index/idc_index_data/__init__.py:10]: Yes, this makes a lot of sense and is a great idea. I would even do `17.0.0`. IDC data releases are coming out at the 2-3 months frequency, and it makes a lot of sense to expect that we may have updates of the index of varying complexity (e.g., adding new tables as opposed to fixing typos or including additional columns in the existing index).

**Comment by jcfr** [comment]
The package `idc-index-data` has been uploaded[^1] (the first release was versioned as `0.1.0`)

Both @fedorov and @vkt1414 have been invited as maintainer on the corresponding PyPI project.

[^1]: https://pypi.org/project/idc-index-data/

**Comment by jcfr** [comment]
@fedorov @vkt1414  Please review and merge at your convenience.

**Review by fedorov — APPROVED** [review]

**Pull request merged**
